### DISCUSSION TITLE
Sandbox name in Chinese characters

### DISCUSSION THREAD

**Discussion by Taeyeonzpx** [sandboxie-plus/Sandboxie#2157] [Various ideas and suggestions]
Can the sandbox name be named in Chinese?????

**Comment by Taeyeonzpx** [comment]
If it is Chinese, will there be garbled code or bugs??

**Comment by bastik-1001** [comment]
Right now it appears to be limited to Latin characters and Arabic numerals (plus underscore), but I am not sure if that is due to it being for practical reasons.

**Reply by DavidXanatos** [reply]
The thing is for the file system unicode should be just fine for the registry as well, but i dont know about the rest of the NT obejct mnager namespace, so
perhaps it would be best to add an option to set a sandbox alias that is used in the display and the ascii only name is hidden?

**Reply by bastik-1001** [reply]
I would not want to trust that allowing unicode for the box name does not mess up something, so that sticking to the current naming model is the best option. Allowing a box name to have an alias, which is used in the UI seems like a good idea. If the UI can display characters in Chinese or Cyrillic, etc. elsewhere, it should be fine to have those characters being used for a box name alias.

Since even the old UI is now available in Korean, that should even work for classic and plus. Though maybe the new UI can have a function for that, so that they don't have to mess with the UI. It seems to be more 'inviting' to those using something other than the Latin alphabet.

**Reply by DavidXanatos** [reply]
So now the question will be about the practial implementation
1. what should be the "physical" box name scheme, we have "DefaultBox" that will stay as is, but how to call additional boxes "Box1" - "Box99", or "Sandbox1" - "Sandbox99" or something else
2. should we switch to creating such boxes by default and only set their aliases? Or should aliases be the exception?

**Reply by bastik-1001** [reply]
I would not change something from how it currently works, since users are used to how it works and Sandboxie does not have to change, so that aliases would be the exception, them only being used and set when the user has a need for it. Mostly those that don't use Latin.

Even for those that use the Latin alphabet, it can be useful to have the box name be an abbreviation or acronym, to keep it short, just to have the UI display something that is more descriptive.

**Reply by diversenok** [reply]
> The thing is for the file system unicode should be just fine for the registry as well, but i don't know about the rest of the NT object manger namespace

File systems and registry are parts of the NT object namespace; of course it supports unicode as well =)

Both registry and object namespace are actually way more permissive than NTFS/FAT/etc.: they allow names to include **any characters except for the back slash**. Which means - all languages, all emoji, forward slash, asterisk, question mask, control ASCII characters (\t, \r, \n, etc.) and even NULL character.

P.S. Have you seen the built-in `HKEY_CURRENT_USER\Control Panel\International\🌎🌏🌍` key? =)

**Reply by DavidXanatos** [reply]
Ok good to know so in principle we can allow any character thats valid for a folder name,
than I really wonder why sandboxie was made to be so restrictive about the box name.

**Reply by diversenok** [reply]
Maybe because it used to break ANSI-versions of Win32 API functions too much? I can understand why programmers used them two decades ago, but there are still tons of new tutorials and code samples that explicitly use `-A` instead of `-W`. At least now we can mitigate it by forcing UTF-8 either [in the application manifest](https://docs.microsoft.com/sr-Latn-RS/windows/win32/sbscs/application-manifests#activecodepage) or globally in regional settings.

**Reply by bastik-1001** [reply]
> Ok good to know so in principle we can allow any character thats valid for a folder name, than I really wonder why sandboxie was made to be so restrictive about the box name.

Maybe you can internally test if Sandboxie does not expect some character set for the box name.

Even with it being limited to 32 characters and the current set there are over 100 octillion possible sandbox names, more than anyone could possibly need. If there was some compatibly reason, the alias is still an option.

**Reply by DavidXanatos** [reply]
I did a quick testing and everything seems to be still working even with a sandbox name like "песочница_ಠ益ಠ" so perhaps we can allow for that in one of the next builds

**Reply by DavidXanatos** [reply]
ideally we should increase the box name as well like double it. And introduce a preprocessor definition holding the langth in one palce, as its now it needs to be changed in like 50 places

**Reply by Taeyeonzpx** [reply]
> ideally we should increase the box name as well like double it. And introduce a preprocessor definition holding the langth in one palce, as its now it needs to be changed in like 50 places

So how to solve this problem? Use alias?

**Reply by bastik-1001** [reply]
If that is going to be changed, it should be tested if right to left override (RTLO) (U+202e) is handled properly, as it may confuse components of sandboxie.

**Comment by 6Tom** [comment]
(The following Chinese is written for Taeyeonzpx. Others please ignore it.）

To Taeyeonzpx: 
你好姐妹，虽然但是，我还是想说，你都有能力上github提问了，还有能力把沙盘用的不错了，你提这问题简直像个外行？utf8的兼容性和ansi有的比？

子目录你随意，兼容写法越多代表开发者越严谨越包容，但是这是顶级根目录，最好兼容性的路径命名方式，肯定是使用全英并且没有空格，这不是常识？微软都没给你让你把盘符命名为中文的权限吧？

就算多搞一步，做个什么名称清单之类，可以只让界面显示为中文，但原路径仍然保持英文，就像翻译一样，但是这有啥意义呢？原路径你还是必须要做各种英文命名啊。

而且有相当一批外国软件，压根就不支持路径含有中文，因为人家就光说英语，人家的世界里压根就没有各种奇奇怪怪的ansi编码，你这不仅是强人所难，而且这样还会使自己的某些软件存在潜在的无法运行的可能性。

而且就你这英文水平，我真的不理解，你图啥？

**Reply by Taeyeonzpx** [reply]
> (The following Chinese is written for Taeyeonzpx. Others please ignore it.）
> 
> To Taeyeonzpx: 你好姐妹，虽然但是，我还是想说，你都有能力上github提问了，还有能力把沙盘用的不错了，你提这问题简直像个外行？utf8的兼容性和ansi有的比？
> 
> 子目录你随意，兼容写法越多代表开发者越严谨越包容，但是这是顶级根目录，最好兼容性的路径命名方式，肯定是使用全英并且没有空格，这不是常识？微软都没给你让你把盘符命名为中文的权限吧？
> 
> 就算多搞一步，做个什么名称清单之类，可以只让界面显示为中文，但原路径仍然保持英文，就像翻译一样，但是这有啥意义呢？原路径你还是必须要做各种英文命名啊。
> 
> 而且有相当一批外国软件，压根就不支持路径含有中文，因为人家就光说英语，人家的世界里压根就没有各种奇奇怪怪的ansi编码，你这不仅是强人所难，而且这样还会使自己的某些软件存在潜在的无法运行的可能性。
> 
> 而且就你这英文水平，我真的不理解，你图啥？

你好 小姐妹 感谢指出我的不足 在你的指点下 学到了很多东西 一定吸取经验  感谢!!!  我也就是个新手   可能以后经验多了 再来看看我现在的提问  也会有你现在的这种心情  说起这个英语水平 也确实非常的惭愧   当初没好好学习英语   在这里在此感谢你的批评  谢谢!

**Reply by 6Tom** [reply]
不你误会了，我这里有几句是反话，引起歧义了，我意思是，你英语水平好，怎么会因为一个路径，纠结起中文来了，感觉你水平全英都ok（中国话才是太难说了😵）

最开始我是无意刷到这问题，感到非常费解，就一掌戳进来了。然后看了你这诉求和水平感觉完全不是一个人，顿时满头大汉，想赶紧跟你分享点我知道的有关情况，语言和口吻组织的都不太好，感谢你能如此宽容，在此补个道歉

还有你性别我是看你头像猜的，然后我赌对了，我是男生😳实在对不住

**Reply by Taeyeonzpx** [reply]
> 不你误会了，我这里有几句是反话，引起歧义了，我意思是，你英语水平好，怎么会因为一个路径，纠结起中文来了，感觉你水平全英都ok（中国话才是太难说了😵）
> 
> 最开始我是无意刷到这问题，感到非常费解，就一掌戳进来了。然后看了你这诉求和水平感觉完全不是一个人，顿时满头大汉，想赶紧跟你分享点我知道的有关情况，语言和口吻组织的都不太好，感谢你能如此宽容，在此补个道歉
> 
> 还有你性别我是看你头像猜的，然后我赌对了，我是男生😳实在对不住

噢噢 好吧  语言表达方式不重要  重要的是你想要分享给我的信息  我理解到了  这点才是最重要的  还是要谢谢你!!
